### PR TITLE
CI: set up linting

### PR DESCRIPTION
**Pull request by JaapWijnen**
Depends on #8 
Same implementation as https://github.com/differentiable-swift/swift-numerics-differentiable/pull/2
except for also adding the `NoAccessLevelOnExtensionDeclaration` rule here. Which we should update in the other repos as well.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Pull Requst
```
Comment (clackary) [inline .github/workflows/pull_request.yml:1]: typo: Request
```suggestion
name: Pull Request
```

**Review by clackary — APPROVED** [review]

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (JaapWijnen) [reply .github/workflows/pull_request.yml]: Fixed in #8 that's a prerequisite of this MR thanks!

**Review by GNMoseke — APPROVED** [review]

**Comment by JaapWijnen** [comment]
I think we'll have to switch to Nick Lockwood's formatter. That way we can also define a custom rule that allows _vjp spelling for functions which I'd like to do. Haven't found a way for custom rules in apple's swift-format yet.

**Comment by JaapWijnen** [comment]
Nice. We should probably add a comment why we're disabling swift-format for those files and link to the issue once we have created one

**Comment by GNMoseke** [comment]
I've overhauled this branch to use Nick's SwiftFormat - PRs coming to other repos as well

In the long term I'd still like to use the apple one just to keep our dependencies completely clean, but it is definitely not ready for primetime yet

**Comment by GNMoseke** [comment]
bump @clackary @JaapWijnen

File: Sources/Differentiation/Dictionary+Differentiation.swift [hunk 2]
```diff
@@ -17,6 +17,14 @@ extension Dictionary: @retroactive Differentiable where Value: Differentiable {
             self[componentKey, default: fatalMissingComponent()].move(by: componentDirection)
         }
     }
+
+    public var zeroTangentVectorInitializer: () -> TangentVector {
```
Comment (JaapWijnen) [inline Sources/Differentiation/Dictionary+Differentiation.swift:21]: This was also removed previously? Did you rebase on main?

Comment (GNMoseke) [reply Sources/Differentiation/Dictionary+Differentiation.swift:21]: I did, must have taken a bad conflict set - lemme fix

**Pull request merged**